pkgs/by-name/bi/binsider/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "binsider";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "orhun";
+    repo = "binsider";
+    rev = "v${version}";
+    hash = "sha256-+QgbSpiDKPTVdSm0teEab1O6OJZKEDpC2ZIZ728e69Y=";
+  };
+
+  cargoHash = "sha256-lXYTZ3nvLrfEgo7AY/qSQYpXsyrdJuQQw43xREezNn0=";
+
+  # Tests need the executable in target/debug/
+  preCheck = ''
+    cargo build
+  '';
+
+  meta = with lib; {
+    description = "Analyzer of executables using a terminal user interface";
+    homepage = "https://github.com/orhun/binsider";
+    license = with licenses; [
+      asl20 # or
+      mit
+    ];
+    maintainers = with maintainers; [ samueltardieu ];
+    mainProgram = "binsider";
+    broken = stdenv.isDarwin || stdenv.isAarch64;
```

Comment:
For my own information, is it better to set `meta.platforms` accordingly or `meta.broken` ?

Reply:
As far as I have understood it, `meta.platforms` is meant to be metadata about which platforms the software claims to support. If upstream doesn't specify that, we set it to what we suspect it supports. That's why we have platforms like `windows` and all sort of others platforms where nix isn't really a thing. If something doesn't work in nixpkgs, but we think upstream supports it (or it *should've* worked if it hadn't been for some nixism), it's the job for `meta.broken`.

However, I've seen `meta.platforms` being set very haphazardly all over the place. Platform groups like `platforms.unix` and `platforms.all` can be very misleading, as they include platforms like `redox` and `ghcjs`. A big cleanup should probably be done at some point, but for now I mostly ignore it as long as nothing explicitly wrong is specified (e.g. if `platforms.windows` or `platforms.unix` when the software is made specifically to interact with the linux kernel)